### PR TITLE
5081 r34 update group member

### PR DESCRIPTION
**Pull request by AnumehaSrivastava05**

### CODE REVIEW

File: frontend/src/views/groupmember/UpdateGroupMember.vue [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import { validateGroupNumber, validatePHN, VALIDATE_PHN_MESSAGE, VALIDATE_GROUP_NUMBER_MESSAGE, VALIDATE_OPTIONAL } from '../../util/validators'
+import { required, helpers } from '@vuelidate/validators'
+
+function validateOptional() {
+  if (this.departmentNumber !== '' || this.groupMemberNumber !== '') {
+    return true;
+  } else {
+    return false;
+  }
+}
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/UpdateGroupMember.vue:48]: I'm not sure this validation works quite as expected. If you don't enter either field then they both show up as required which isn't technically correct. I think this validation might be best handled manually. E.g. "At least one of Group Member or Department Number is required" See PhnInquiry.vue.

File: frontend/src/services/GroupMemberService.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { apiRequest, resources } from './BaseService'
+
+export default {
+
+  updateGroupMember(request) { 
+   console.log(`Update Empoyee: [PHN: ${request.phn}] [Group Number: ${request.groupNumber}][DEPARTMENT: ${request.departmentNumber}] [Group Member Number: ${request.groupMemberNumber}]`)
+   return apiRequest().then(axiosInstance => axiosInstance.post(resources.groupMember.updateGroupMember, request)) 
+  },
+
+
```
Comment (weskubo-cgi) [inline frontend/src/services/GroupMemberService.js:10]: Extra whitespace not required.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]
I believe Dave B will have a look as well.

File: frontend/src/views/groupmember/UpdateGroupMember.vue [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  validations() {
+    return {
+      phn: {
+        required,
+        validatePHN: helpers.withMessage(
+          VALIDATE_PHN_MESSAGE, validatePHN
+        )
+      },
+      groupNumber: { required,
+      validateGroupNumber: helpers.withMessage(
```
Comment (weskubo-cgi) [inline frontend/src/views/groupmember/UpdateGroupMember.vue:133]: Formatting is inconsistent: required should be on next line and validateGroupNumber should be indented.

**Review by daveb-hni — COMMENTED** [review]
Minor API suggestions, otherwise everything covered for UI

File: frontend/src/util/validators.js [hunk 4]
```diff
@@ -177,3 +177,4 @@ export const VALIDATE_GROUP_NUMBER_MESSAGE = "Group Number is invalid"
 export const VALIDATE_GROUP_MEMBER_NUMBER_MESSAGE = "Group Member Number is invalid"
 export const VALIDATE_DEPARTMENT_NUMBER_MESSAGE = "Department Number is invalid"
 export const VALIDATE_TELEPHONE_MESSAGE = "Telephone is invalid. Only numbers 0 to 9 are valid. Phone Number must be entered as ten (10) numbers in length with no space or hyphen."
+export const VALIDATE_OPTIONAL = "Field is required"
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/util/validators.js:180]: I will remove this.

File: frontend/src/router/index.js [hunk 5]
```diff
@@ -1,19 +1,21 @@
...
-import store from '../store'
-import NotFound from '../views/NotFound.vue'
 import AddVisaResidentWithPHN from '../views/coverage/enrollment/AddVisaResidentWithPHN.vue'
 import AddVisaResidentWithoutPHN from '../views/coverage/enrollment/AddVisaResidentWithoutPHN.vue'
+import CheckEligibility from './../views/eligibility/CheckEligibility.vue'
 import CoverageEnrollmentHome from '../views/coverage/enrollment/CoverageEnrollmentHome.vue'
 import CoverageMaintenanceHome from '../views/coverage/maintenance/CoverageMaintenanceHome.vue'
+import CoverageStatusCheck from './../views/eligibility/CoverageStatusCheck.vue'
 import EligibilityHome from '../views/eligibility/EligibilityHome.vue'
+import Employees from './../views/Employees.vue'
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/router/index.js]: I have removed the file.

File: frontend/src/components/template/TheNavBar.vue [hunk 6]
```diff
@@ -29,7 +29,12 @@
           </div>
         </li>
         <li id="manage-employees-link" :class="tabClass($route, 'ManageEmployees')">
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/components/template/TheNavBar.vue]: The argument passed in should be a path, not the component name. I have fixed this.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/GroupMemberController.java [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+				RPBSPED0 ped0Request = rpbsped0Converter.convertRequest(updateNumberAndDeptRequest);
+				RPBSPED0 ped0Response = groupMemberService.updateGroupMemberDepartmentNumber(ped0Request);
+				deptNumberResponse = rpbsped0Converter.convertResponse(ped0Response);
+			}
+
+			// Handle the group member/employee number
+			UpdateNumberAndDeptResponse empNumberResponse = new UpdateNumberAndDeptResponse();
+			if (StringUtils.isNotBlank(updateNumberAndDeptRequest.getGroupMemberNumber())) {
+				RPBSPEE0Converter rpbspee0Converter = new RPBSPEE0Converter();
+				RPBSPEE0 pee0Request = rpbspee0Converter.convertRequest(updateNumberAndDeptRequest);
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/GroupMemberController.java:72]: Fixed.

File: backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/GroupMemberController.java [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+			response.setStatus(StatusEnum.SUCCESS);
+		}
+		response.setPhn(StringUtils.isNotBlank(deptNumberResponse.getPhn()) ? deptNumberResponse.getPhn() : empNumberResponse.getPhn());
+		return response;		
+	}
+	
+	private String generateErrorWarningMessage(UpdateNumberAndDeptResponse deptNumberResponse, UpdateNumberAndDeptResponse empNumberResponse) {
+		Set<String> messages = new HashSet<>();
+		
+		// Include all Error and Warning messages but
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply backend/src/main/java/ca/bc/gov/hlth/hnweb/controller/GroupMemberController.java:120]: Fixed.

**Pull request merged**